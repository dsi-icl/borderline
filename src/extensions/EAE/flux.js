import NavigationButton from './containers/NavigationButton'
import View from './containers/View'

export const actions = {

    dockToPager: () => ({
        type: '@@core/pager/PAGE_DOCK',
        path: 'eae',
        icon: NavigationButton,
        view: View
    })
};

export const epics = {

    enclaveBoot:
    (action) => action.ofType('START')
        .mapTo(actions.dockToPager()),

};

export const reducers = {
}

export default {
    actions,
    epics,
    reducers
};
