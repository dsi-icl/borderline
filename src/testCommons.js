import React from 'react';
import renderer from 'react-test-renderer';

export const canRender = (Component) => {

    it('renders without crashing while empty', () => {
        const tree = renderer.create(<Component />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders without crashing with a child', () => {
        const tree = renderer.create(<Component>
            <h1>Hello World</h1>
        </Component>).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders without crashing with multiple children', () => {
        const tree = renderer.create(<Component>
            <h1>Hello</h1>
            <h1>World</h1>
        </Component>).toJSON();
        expect(tree).toMatchSnapshot();
    });
}
