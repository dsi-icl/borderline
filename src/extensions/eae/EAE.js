import { Component } from 'react';
import { epics, reducers } from './flux';

class EAE extends Component {

    // Custom name for container
    static displayName = 'EAE';

    // Custom properties for borderline model
    static modelName = 'eae';
    static modelEpics = epics;
    static modelReducers = reducers;

    render() {
        return null;
    }
}

export default EAE;
