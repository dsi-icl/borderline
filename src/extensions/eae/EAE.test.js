import EAE from './EAE';
import { canRender } from 'testCommons';

canRender(EAE);
