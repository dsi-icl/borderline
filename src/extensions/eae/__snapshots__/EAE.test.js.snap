// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing while empty 1`] = `null`;

exports[`renders without crashing with a child 1`] = `
<h1>
  Hello World
</h1>
`;

exports[`renders without crashing with multiple children 1`] = `
Array [
  <h1>
    Hello
</h1>,
  <h1>
    World
</h1>,
]
`;
